src/mscorlib/src/System/Threading/ThreadPool.cs

```diff
@@ -491,6 +491,9 @@ public IThreadPoolWorkItem Dequeue(ThreadPoolWorkQueueThreadLocals tl, ref bool
 
         internal static bool Dispatch()
         {
+            // Put threadpool thread on default context
+            ExecutionContextSwitcher.SetDefaults();
```

Comment:
If its `null` then it needs to do a memory barrier assign to `Default`; so its once per dispatch or once per loop.